syntax = "proto3";

package blockchain;

message Status {
    bytes hash = 1;
    uint64 height = 2;
}

enum Crypto {
    DEFAULT = 0;
    RESERVED = 1;
}

message Transaction {
    string to = 1;
    string nonce = 2;
    uint64 quota = 3;
    uint64 valid_until_block = 4;
    bytes data = 5;
    bytes value = 6;
    uint32 chain_id = 7;
    uint32 version = 8;
    bytes to_v1 = 9;
    bytes chain_id_v1 = 10;
}

message UnverifiedTransaction {
    Transaction transaction = 1;
    bytes signature = 2;
    Crypto crypto = 3;
}

message SignedTransaction {
    UnverifiedTransaction transaction_with_sig = 1;
    // SignedTransaction hash
    bytes tx_hash = 2;
    // public key
    bytes signer = 3;
}
